shared/components/FormField/FormField.jsx

```diff
@@ -88,13 +88,18 @@ class FormField extends React.Component {
   onChange = event => {
     const { onChange } = this.props
 
-    this.setState({
-      value: event.target.value,
-    })
-
-    if (onChange) {
-      onChange(event)
-    }
+    event.persist()
```

Comment:
Oh yeah, thats an interesting thing. Here is the official docs on `event.persist()` --> https://reactjs.org/docs/events.html#event-pooling

Basically, once the callback finishes (the end of the `Input.onChange` function), the event cannot be accessed anymore. Since the `setState` callback gets called asynchronously, it happens AFTER `Input.onChange`, so the event would be nullified by then unless you persist it. 

Honestly I'm not totally sure why, but has to do with React's synthetic event implementation